retrofit/src/main/java/retrofit/RequestFactoryParser.java

```diff
@@ -155,15 +170,17 @@ private void parseHttpMethodAndPath(String httpMethod, String value, boolean has
     this.httpMethod = httpMethod;
     this.hasBody = hasBody;
 
-    if (value.isEmpty()) {
+    String combinedRelativeUrl = getPathPrefix() + value;
+
+    if (combinedRelativeUrl.isEmpty()) {
       return;
     }
 
     // Get the relative URL path and existing query string, if present.
-    int question = value.indexOf('?');
-    if (question != -1 && question < value.length() - 1) {
+    int question = combinedRelativeUrl.indexOf('?');
+    if (question != -1 && question < combinedRelativeUrl.length() - 1) {
       // Ensure the query string does not have any named parameters.
-      String queryParams = value.substring(question + 1);
+      String queryParams = combinedRelativeUrl.substring(question + 1);
```

Comment:
@RoyMontoya I'm not sure if this PR is the right place to discuss this. 
Would the problem you are describing here be introduced by this PR, or does it already exist in master?